src/NHibernate/Dialect/Dialect.cs

```diff
@@ -274,7 +274,7 @@ public virtual string GetCastTypeName(SqlType sqlType)
 		/// length (if appropriate)
 		/// </summary>
 		/// <param name="code">The typecode</param>
-		/// <param name="capacity">Maximum length of database type</param>
+		/// <param name="capacity">Maximum length or scale of database type</param>
 		/// <param name="name">The database type name</param>
 		protected void RegisterColumnType(DbType code, int capacity, string name)
```

Comment:
In fact it appears `capacity` is also used for transmitting max precision for `decimal` types, but the retrieval code does not switch to precision for it. Since the retrieval code would switch to default in case of over capacity, which in the case of `decimal` is far from the maximum, that seems better it does not, although not very satisfactory.

Moreover, current `capacity` values transmitted for decimal are currently plain wrong, at least for sql server: `19`, which neither match its maximal precision `38` (at least since 2000, although some current docs write without much precision that it was `28` in some earlier version) nor its maximal size `17`.